src/app/frontend/common/resource/editresource_controller.js

```diff
@@ -67,11 +75,35 @@ export class EditResourceController {
   }
 
   /**
+   * @export
+   */
+  copy() {
+    /* if the clipboard functionality is not supported */
+    if (!this.clipboard_.supported) {
+      this.showMessage('Your browser does not support clipboard');
```

Comment:
All messages should be localized. You can find usage examples in other files. Look for `goog.getMsg` calls.